mne/inverse_sparse/mxne_optim.py

```diff
@@ -339,30 +336,30 @@ def _mixed_norm_solver_bcd(M, G, alpha, lipschitz_constant, maxit=200,
 
     alpha_lc = alpha / lipschitz_constant
 
-    for i in range(maxit):
-        for j in range(n_positions):
-            idx = slice(j * n_orient, (j + 1) * n_orient)
-
-            G_j = G[:, idx]
-            X_j = X[idx]
-
-            X_j_new = np.dot(G_j.T, R) / lipschitz_constant[j]
+    G = np.asfortranarray(G)
+    # It is better to call gemm and gemm2 here
+    # so it is called only once
+    # gemm and gemm2 are 2 functions (callable) for matrix multiplication
+    # Two different functions are called because
+    # matrix multiplications are performed on matrices with different shapes
+    gemm = linalg.get_blas_funcs("gemm", [R.T, G[:, 0:n_orient]])
+    gemm2 = linalg.get_blas_funcs("gemm", [X[0:n_orient, :].T,
+                                  np.asfortranarray(G[:, 0:n_orient].T)])
```

Comment:
Did you see my comment before? I suspect that because R and X are C cont and you do .T here the inputs will all be F cont, so you should have `gemm is gemm2`. Can you check? If it isn't the case, why are they not the same? (All that should matter is dtypes, which should all be double here, and the FORTRAN flag, which should be true for all inputs here.)

Reply:
You are these functions are the same, I will remove gemm2